src/main/java/org/folio/rest/impl/LookupsUtils.java

```diff
@@ -1,68 +1,190 @@
       .thenApply(LookupsUtils::verifyAndExtractBody);
   }
 
-  static CompletableFuture<JsonObject> getUser(String userId, Map<String, String> okapiHeaders,
-                                               HttpClientInterface httpClient) {
-    return get("/users/" + userId, httpClient, okapiHeaders)
+  static CompletableFuture<JsonObject> getUser(String userId, Map<String, String> okapiHeaders) {
+    return get("/users/" + userId, okapiHeaders)
       .thenApply(LookupsUtils::verifyAndExtractBody);
   }
 
-  static CompletableFuture<JsonObject> getRequestPolicyId(String queryString, Map<String, String> okapiHeaders,
-                                                          HttpClientInterface httpClient) {
-    return get("/circulation/rules/request-policy?" + queryString, httpClient, okapiHeaders)
+  static CompletableFuture<JsonObject> getRequestPolicyId(RequestTypeParameters criteria, Map<String, String> okapiHeaders) {
+    final var queryParameters = Map.of(
+      "item_type_id", criteria.getItemMaterialTypeId(),
+      "loan_type_id", criteria.getItemLoanTypeId(),
+      "patron_type_id", criteria.getPatronGroupId(),
+      "location_id", criteria.getItemLocationId());
+
+    return get("/circulation/rules/request-policy",
+        queryParameters, okapiHeaders)
       .thenApply(LookupsUtils::verifyAndExtractBody);
   }
 
-  static CompletableFuture<JsonObject> getRequestPolicy(String requestPolicyId, Map<String, String> okapiHeaders,
-                                                        HttpClientInterface httpClient) {
-    return get("/request-policy-storage/request-policies/" + requestPolicyId, httpClient, okapiHeaders)
+  static CompletableFuture<JsonObject> getRequestPolicy(String requestPolicyId, Map<String, String> okapiHeaders) {
+    return get("/request-policy-storage/request-policies/" + requestPolicyId, okapiHeaders)
       .thenApply(LookupsUtils::verifyAndExtractBody);
   }
 
   static JsonObject verifyAndExtractBody(Response response) {
-    if (!Response.isSuccess(response.getCode())) {
-      throw new CompletionException(new HttpException(response.getCode(),
-        response.getError().getString("errorMessage")));
+    if (!response.isSuccess()) {
+      throw new CompletionException(new HttpException(response.statusCode,
+        response.body));
+    }
+
+    // Parsing an emppty body to JSON causes an exception
+    if (StringUtils.isBlank(response.body)) {
+      return null;
     }
 
-    return response.getBody();
+    return new JsonObject(response.body);
   }
 
-  static HttpClientInterface getHttpClient(Map<String, String> okapiHeaders) {
-    final String okapiURL = okapiHeaders.getOrDefault("X-Okapi-Url", "");
-    final String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
+  public static CompletableFuture<LookupsUtils.Response> post(String path,
+    JsonObject body, Map<String, String> okapiHeaders) {
```

Comment:
there's a lot of shared code between the post and put methods-is it maybe worth thinking about a shared utility function they both use?

Reply:
Yes, it likely is. I think there is a lot of follow up work needed to improve the quality of this code. I didn't think we had time to do it in this pull request.